examples/infinite_scroll/infinite_scroll.py

```diff
@@ -0,0 +1,36 @@
+import logging
+
+import agentql
+from playwright.sync_api import sync_playwright
+
+logging.basicConfig(level=logging.DEBUG)
+log = logging.getLogger(__name__)
+
+
+if __name__ == "__main__":
+    QUERY = """
+    {
+        page_title
+        post_headers[]
+    }
+    """
+    with sync_playwright() as playwright, playwright.chromium.launch(headless=False) as browser:
+        page = agentql.wrap(browser.new_page())
+
+        log.info("Navigating to the page...")
+
+        page.goto("https://infinite-scroll.com/demo/full-page/")
+        page.wait_for_page_ready_state()
+
+        num_extra_pages_to_load = 3
+
+        for times in range(num_extra_pages_to_load):
+            log.info(f"Scrolling to the bottom of the page... (num_times = {times+1})")
+            page.keyboard.press("End")
```

Comment:
I think this should work in most cases but may result in unexpected behaviors in some rare cases. One example would be the Youtube page, I think pressing the `end` key will actually interact with the video progress bar instead of scrolling the page. To reliably scroll to bottom, we might need to rely on info like `totalHeight`and `viewportHeight`. This is how we used to do it:

```
self.wait_for_page_ready_state()
        height_information = self.current_page.evaluate(load_js("get_scroll_info"))
        viewport_height, total_height, scroll_height = height_information
        while scroll_height < total_height:
            scroll_height = scroll_height + viewport_height
            self.current_page.mouse.wheel(delta_x=0, delta_y=viewport_height)
            time.sleep(random.uniform(0.05, 0.1))
```

Reply:
oh, interesting.  that's a bit more complex to explain in the docs tutorial.  i do try to caveat that this solution doesn't work for all cases, but the tradeoff is a more complex example that may solve some known issues already.  let me noodle on this a bit!